### ISSUE TITLE
A problem when running bismark using the test_R1.fastq.gz、test_R2.fastq.gz

### ISSUE THREAD

**Issue by hazel914** [FelixKrueger/Bismark#624]
Dear Bismark author,
It's my first time to use the bismark. And i'm facing a problem even when i use the test data you have provided: test_R1.fastq.gz, test_R2.fastq.gz. And I have checked the bowtie2 and samtools have already been available in the PATH environment.

Here's my code:
`/home/xiaojuan/anaconda3/bin/bismark --bowtie2 -N 1 --unmapped --non_directional --maxins 10000 --genome_folder /home/dell/reference_genome/human/hg38/ --output_dir /home/xiaojuan/meth_test/03.bismark_data/H1-1/test3  -1 test_R1.fastq.gz  -2 test_R2.fastq.gz
`

and it's log output shows an unexpected stop when:
`>>> Writing bisulfite mapping results to test_R1_bismark_bt2_pe.bam <<< `
`Unmapped sequences will be written to test_R1.fastq.gz_unmapped_reads_1.fq.gz and test_R2.fastq.gz_unmapped_reads_2.fq.gz
`

The full attachment log file is here: 
[bismark.align.test.log](https://github.com/FelixKrueger/Bismark/files/12574652/bismark.align.test.log)

Thank you!
Best regards,
hazel

**Comment by FelixKrueger** [comment]
Hi Hazel,

there is nothing fundamentally wrong with your command (see below), the log file looks as expected, apart from the fact that it didn't finish. My suspicion is that the process got killed because you ran out of resources, is that a possibility? What are your system resources?

### Here are a few comments about options you selected 

- `--bowtie2`: this is the default, no need to specify
- `-N 1`: I would generally advise not to use this as it will make the compute time a lot longer and not increase the number of alignments notably for most cases
- `--non_directional`: The test data set it directional, so you should not be using this flag. Also, it will use two additional alignment threads, which means it will generally take longer, and for the human genome probably use another 6-8GB of RAM (in addition to being the wrong flag for the test data). In case you are running Bismark on a machine with low system resources, this flag might cause the process to get killed for memory reasons
- `--maxins 10000`: generally, bisulfite sequencing fragments tend to be quite short (often the vast majority being in the range of 50-150bp). Selecting a value of 10,000 will just increase alignments times a fair bit, without any measurable benefit

Generally, I would try to stick to the default parameters (after all there is a reason for the defaults...) :)

**Comment by hazel914** [comment]
Hi Felix, thank you so much for your fast reply and useful suggestions!

I have solved this problem and runned Bismark successfully while using the test files. It turns out that there's someting wrong with my samtools, it didn't work because of an error: `error while loading shared libraries: libcrypto.so.1.0.0: cannot open shared object file: No such file or directory`. 

After i reinstalled the samtools. this problem was solved, and Bismark works well! 

And i'm also thinking about how this problem may occured. I think it maybe because that I recently started to use this new server, so the samtools may was installed while i'm using `conda install Bismark`. And maybe this contributed to the lib problems with samtools? And i think  it may be useful if bismark can report these errors in the log file if there's something wrong with its' needed tools.

Best regards,
hazel
